nixos/modules/services/monitoring/timescale-prometheus.nix

```diff
@@ -0,0 +1,262 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.timescale-prometheus;
+
+in
+{
+
+  options = {
+    services.timescale-prometheus = {
+
+      enable = mkEnableOption "timescale prometheus service";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.timescale-prometheus;
+        defaultText = "pkgs.timescale-prometheus";
+        description = "The timescale-prometheus package to use.";
+      };
+
+      async-acks = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
+          Ack before data is written to DB [TS_PROM_ASYNC_ACKS]
+        '';
+      };
+
+      db-connect-retries = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          How many times to retry connecting to the database [TS_PROM_DB_CONNECT_RETRIES]
+        '';
+      };
+
+      db-host = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = ''
+          The TimescaleDB host [TS_PROM_DB_HOST] (default "localhost")
+        '';
+      };
+
+      db-name = mkOption {
+        type = types.str;
+        default = "timescale";
+        description = ''
+          The TimescaleDB database [TS_PROM_DB_NAME] (default "timescale")
+        '';
+      };
+
+      db-password = mkOption {
```

Comment:
I'm not a fan of passwords in the nix store. We shouldn't do this. :-1:

Reply:
Neither am I. Is this implementation sufficient or have I missed something from passwordFile best practice?

Reply:
Unfortunately your new `db-passwordFile` option still ends up in the nix store.

Reply:
Thanks for pointing this out.
I wonder if my head is now able to come up with a solution.